src/Vts/IO/CustomBinaryStreamWriterOfT.cs

```diff
@@ -10,11 +11,11 @@ namespace Vts.IO
     /// <typeparam name="T">Type of the data</typeparam>
     public class CustomBinaryStreamWriter<T> : IDisposable
     {
-        private string _filename;
-        private Action<BinaryWriter, T> _writeMap;
-
-        private Stream _stream;
-        private BinaryWriter _binaryWriter;
+        private readonly string _filename;
+        private readonly Action<BinaryWriter, T> _writeMap;
+        private Stream? _stream;
+        private BinaryWriter? _binaryWriter;
+        private bool disposedValue;

```

Comment:
This name should be _disposedValue - if you merge before seeing this, we can fix later.